csxtools/utils.py

```diff
@@ -76,20 +77,28 @@ def get_fastccd_images(light_header, dark_headers=None,
 
         bgnd = np.array(dark)
 
-    data = _get_images(light_header, tag)
-    data = correct_images(data, bgnd, gain=gain)
+    data, n_images = _get_images(light_header, tag)
+    data = correct_images(data, bgnd, flat=flat, gain=gain)
     data = rotate90(data, 'cw')
-    return data
+    return data, n_images
 
 def _get_images(header, tag):
     t = ttime.time()
-    images = get_images(header, tag)[0]
+
+    all_event_images = get_images(header, tag)
+
+    images = []
+    n_images = []
+    for event_images in all_event_images:
+        images.append(np.asarray(event_images, np.uint16))
+        n_images.append(len(images[-1]))
```

Comment:
@stuwilkins @danielballan This is to easily group images by event if needed. I've often found it very useful. For instance, if you want to plot total detector intensity (or that of some ROI) vs some variable, such as energy, where each point is an event having a number of images, you can use a simple function like this:

images, n_images = get_fastccd_images(db....)
N = np.insert(n_images, 0, 0)
images_by_event = [images[n1:n2] for n1, n2 in zip(N[:-1], N[1:])]
ccd_tot = [np.nanmean(images) for images in images_by_event]

and so on..

Maybe there are better ways of doing this. But this works pretty well for me.